__tests__/VcIssuanceClient.spec.ts

```diff
@@ -6,53 +9,84 @@ import {
   CredentialResponse,
   CredentialResponseError,
   CredentialResponseErrorCode,
-  ProofType
+  JWTSignerArgs, ProofOfPossession,
 } from '../src/main/types';
 
+const partialJWT = "eyJhbGciOiJSUzI1NiIsImtpZCI6ImRpZDpleGFtcGxlOmViZmViMWY3MTJlYmM2ZjFjMjc2ZTEyZWMyMS9rZXlzLzEifQ.eyJpc3MiOiJzNkJoZFJrcXQzIiwiYXVkIjoiaHR0cHM6Ly9zZXJ2ZXIuZXhhbXBsZS5jb20iLCJpYXQiOjE2NTkxNDU5MjQsIm5vbmNlIjoidFppZ25zbkZicCJ9"
+
+const signJWT = async (args: JWTSignerArgs): Promise<string> => {
+  const {header, payload, privateKey} = args
+  return await new jose.SignJWT({...payload})
+  .setProtectedHeader({...header, alg: 'RS256'})
+  .sign(privateKey)
+}
+
+const jwtArgs: JWTSignerArgs = {
+  header: {
+    alg: "RS256",
+    kid: "did:example:ebfeb1f712ebc6f1c276e12ec21/keys/1"
+  },
+  payload: {
+    iss: "s6BhdRkqt3",
+    aud: "https://server.example.com",
+    iat: 1659145924,
+    nonce: "tZignsnFbp"
+  },
+  privateKey: undefined
+}
+
+
+beforeAll(async () => {
+  jwtArgs.privateKey = generateKeyPairSync("rsa", {
+    modulusLength: 4096
+  }).privateKey
+})
+
 describe('VcIssuanceClient ', () => {
   it('should build correctly provided with correct params', function () {
     const vcIssuanceClient = VcIssuanceClient.builder()
-      .withCredentialRequestUrl('https://oidc4vci.demo.spruceid.com/credential')
-      .withFormat('jwt_vc')
-      .build();
+    .withCredentialRequestUrl('https://oidc4vci.demo.spruceid.com/credential')
+    .withFormat('jwt_vc')
+    .build();
     expect(vcIssuanceClient._issuanceRequestOpts.credentialRequestUrl).toBe('https://oidc4vci.demo.spruceid.com/credential');
   });
 
-  it('should build credential request correctly', function () {
+  it('should build credential request correctly', async () => {
     const vcIssuanceClient = VcIssuanceClient.builder()
-      .withCredentialRequestUrl('https://oidc4vci.demo.spruceid.com/credential')
-      .withFormat('jwt_vc')
-      .withCredentialType('https://imsglobal.github.io/openbadges-specification/ob_v3p0.html#OpenBadgeCredential')
-      .build();
-    const credentialRequest: CredentialRequest = vcIssuanceClient.createCredentialRequest({
-      proof: {
-        proof_type: ProofType.JWT,
-        jwt: 'eyJraWQiOiJkaWQ6ZXhhbXBsZTplYmZlYjFmNzEyZWJjNmYxYzI3NmUxMmVjMjEva2V5cy8xIiwiYWxnIjoiRVMyNTYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJzNkJoZFJrcXQzIiwiYXVkIjoiaHR0cHM6Ly9zZXJ2ZXIuZXhhbXBsZS5jb20iLCJpYXQiOiIyMDE4LTA5LTE0VDIxOjE5OjEwWiIsIm5vbmNlIjoidFppZ25zbkZicCJ9.ewdkIkPV50iOeBUqMXCC_aZKPxgihac0aW9EkL1nOzM',
-      }
-    });
+    .withCredentialRequestUrl('https://oidc4vci.demo.spruceid.com/credential')
+    .withCredentialRequestUrl('oidc4vci.demo.spruceid.com/credential')
```

Comment:
Why does it have 2 requestUrls of which one is incorrect?

Reply:
Merge problem, fixed